pandas/io/json/json.py

```diff
@@ -171,11 +203,19 @@ def __init__(self, obj, orient, date_format, double_precision,
         self.obj = obj.reset_index()
         self.date_format = 'iso'
         self.orient = 'records'
-
-    def write(self):
-        data = super(JSONTableWriter, self).write()
+        self.index = index
+
+    def _write(self, obj, orient, double_precision, ensure_ascii,
+               date_unit, iso_dates, default_handler):
+        if not self.index:
+            obj = obj.drop('index', axis=1)
```

Comment:
this is not robust (you can have a MultiIndex, or an index with another name, etc ..) 
You need to handle this above when `reset_index` is called (do `drop=True/False` depending on the value of `self.index`)

(and also make sure to add a test for this)

Reply:
Thanks, that's a good point. I think I have it fixed now.

Reply:
Did you push the new commits? (I don't see any change here)

Reply:
Yeah sorry I forgot to push the new commit. I've pushed now.

Reply:
That looks better now!